src/main/scala/com/cognite/sdk/scala/common/ReadableResource.scala

```diff
@@ -7,69 +7,99 @@ import fs2._
 
+// TODO: Verify that index and numPartitions are valid
+final case class Partition(index: Int = 1, numPartitions: Int = 1) {
+  override def toString: String = s"$index/$numPartitions"
+}
+
 trait Readable[R, F[_]] extends WithRequestSession[F] with BaseUri {
-  def readWithCursor(cursor: Option[String], limit: Option[Long]): F[ItemsWithCursor[R]]
+  def readWithCursor(
+      cursor: Option[String],
+      limit: Option[Long],
+      partition: Partition = Partition()
+  ): F[ItemsWithCursor[R]]
 
-  def readFromCursor(cursor: String): F[ItemsWithCursor[R]] =
-    readWithCursor(Some(cursor), None)
+  def readFromCursor(cursor: String, partition: Partition = Partition()): F[ItemsWithCursor[R]] =
+    readWithCursor(Some(cursor), None, partition)
 
-  def readFromCursorWithLimit(cursor: String, limit: Long): F[ItemsWithCursor[R]] =
-    readWithCursor(Some(cursor), Some(limit))
+  def readFromCursorWithLimit(
+      cursor: String,
+      limit: Long,
+      partition: Partition = Partition()
+  ): F[ItemsWithCursor[R]] =
+    readWithCursor(Some(cursor), Some(limit), partition)
 
-  def read(): F[ItemsWithCursor[R]] = readWithCursor(None, None)
+  def read(partition: Partition = Partition()): F[ItemsWithCursor[R]] =
+    readWithCursor(None, None, Partition())
 
-  def readWithLimit(limit: Long): F[ItemsWithCursor[R]] =
-    readWithCursor(None, Some(limit))
+  def readWithLimit(limit: Long, partition: Partition = Partition()): F[ItemsWithCursor[R]] =
+    readWithCursor(None, Some(limit), partition)
 
   private def listWithNextCursor(
       cursor: Option[String],
-      limit: Option[Long]
+      limit: Option[Long],
+      partition: Partition
   ): Stream[F, R] =
-    Readable.pullFromCursorWithLimit(cursor, limit, readWithCursor).stream
+    Readable.pullFromCursorWithLimit(cursor, limit, partition, readWithCursor).stream
 
-  def listFromCursor(cursor: String): Stream[F, R] =
-    listWithNextCursor(Some(cursor), None)
+  def listFromCursor(cursor: String, partition: Partition = Partition()): Stream[F, R] =
+    listWithNextCursor(Some(cursor), None, partition)
 
-  def listWithLimit(limit: Long): Stream[F, R] =
-    listWithNextCursor(None, Some(limit))
+  def listWithLimit(limit: Long, partition: Partition = Partition()): Stream[F, R] =
+    listWithNextCursor(None, Some(limit), partition)
 
-  def listFromCursorWithLimit(cursor: String, limit: Long): Stream[F, R] =
-    listWithNextCursor(Some(cursor), Some(limit))
+  def listFromCursorWithLimit(
+      cursor: String,
+      limit: Long,
+      partition: Partition = Partition()
+  ): Stream[F, R] =
+    listWithNextCursor(Some(cursor), Some(limit), partition)
+
+  def list(partition: Partition = Partition()): Stream[F, R] =
+    listWithNextCursor(None, None, partition)
 
-  def list(): Stream[F, R] = listWithNextCursor(None, None)
+  def listPartitions(numPartitions: Int): Seq[Stream[F, R]] =
+    Seq(listWithNextCursor(None, None, Partition()))
```

Comment:
Is the point here to let `listPartitions` work for resources without partitioning? We could also hide it from those resources by adding a trait or the like, but perhaps that would give us other problems?

Reply:
Yes, but it might be better to support it as a separate trait. That way we can guarantee that the sequence of streams returned has exactly `numPartitions` items.